core/rubygems/version.rbs

```diff
@@ -190,7 +190,7 @@ module Gem
     # Constructs a Version from the `version` string.  A version string is a series
     # of digits or ASCII letters separated by dots.
     #
-    def initialize: (_ToS version) -> void
+    def initialize: (_ToS & Kernel version) -> void
```

Comment:
`nil?` and `is_a?` (https://github.com/ruby/ruby/blob/9ee58b2054c1bbe722ae5a2a4ec6a750ee583220/lib/rubygems/version.rb#L221)